Update mupdf to 1.8

ericwa opened the pull request:
Besides updating the submodule commit:
- small update to the mupdf patch we apply (`fz_paint_image_with_color` got an extra argument)
- had to update `document-viewer/jni/mupdf/ThirdParty.mk` due to a file being renamed in Freetype
- change most `fz_free` calls in `document-viewer/jni/ebookdroid/mupdfdroidbridge.c` to the proper `fz_drop_*` call. Was getting crashes when the code tried to `fz_free` an `fz_outline`. The remaining `fz_free()` calls are for `renderpage_t` structures that are allocated in `mupdfdroidbridge.c` with `fz_malloc_no_throw`, so it should be correct to `fz_free` those.

In addition, (possibly undesirable) I had to remove `LOCAL_CFLAGS += -DNOCJK` from Core.mk, or else epubs failed to load. I reported this to the mupdf bugtracker: http://bugs.ghostscript.com/show_bug.cgi?id=696326
NOCJK was also discussed here: https://github.com/SufficientlySecure/document-viewer/issues/115

tuxor1337 commented:
Hm, I wouldn't upgrade to 1.8 as long as we have the problem about the NOCJK flag. Or are there really critical fixes or fantastic new features in 1.8?

ericwa commented:
Agree, let's not merge this for now, I don't think there is anything critical in 1.8: http://mupdf.com/news

ericwa commented:
Sounds like the broken NOCJK option in mupdf was fixed: http://bugs.ghostscript.com/show_bug.cgi?id=696326
so we can update when the next mupdf release comes out, or to a git version.

timofonic commented:
So MuPDF 1.8 will be available?

Any news about ePub3 support in the library? I seems to be under development, but not sure if it will be available soon or not. Currently, there's no FOSS epub3 that isn't based on a web browser.

ericwa commented:
It looks like a small patch to fix the NOCJK option in MuPDF 1.8:
http://git.ghostscript.com/?p=mupdf.git;a=commitdiff;h=3116d629d72ff7bf227a46601d97db8d283c0650

We could consider upgrading to 1.8 with this patch, assuming it works ok?

ericwa commented:
I rebased this branch onto current master to make testing easier, and added the patch from http://git.ghostscript.com/?p=mupdf.git;a=commitdiff;h=3116d629d72ff7bf227a46601d97db8d283c0650 , and re-enabled -NOCJK. 

Everything seems OK so far, I'll test it a bit more.
